packages/cerebral/src/Module.js

```diff
@@ -1,35 +1,24 @@
 class Module {
-  constructor (path, module, controller) {
-    this.name = path.slice().pop()
-    this.path = path
-    this.controller = controller
-    this.modules = {}
+  constructor (controller, path, moduleDescription) {
+    const moduleStub = {
+      controller,
+      path: path.join('.'),
+      name: path.slice().pop()
+    }
 
-    const moduleDescription = typeof module === 'function' ? module(this) : module
+    const module = typeof moduleDescription === 'function' ? moduleDescription(moduleStub) : moduleDescription
 
-    this.initialState = moduleDescription.state || {}
-    if (moduleDescription.state) {
-      this.controller.getModel().set(this.path, moduleDescription.state)
-    }
+    module.initialState = module.state || {}
+    controller.getModel().set(path, module.initialState)
 
-    this.signals = moduleDescription.signals || {}
-    this.provider = moduleDescription.provider
-    this.registerModules(moduleDescription.modules || {})
-  }
-  registerModules (modules) {
-    Object.keys(modules).forEach(moduleKey => {
-      this.modules[moduleKey] = new Module(this.path.concat(moduleKey), modules[moduleKey], this.controller)
-    })
-  }
-  /*
-    Used by controller to extract all the providers of all modules
-  */
-  getProviders () {
-    return (this.provider ? [this.provider] : []).concat(Object.keys(this.modules)
-      .reduce((nestedProviders, moduleKey) => {
-        return nestedProviders.concat(this.modules[moduleKey].getProviders())
-      }, [])
-    )
+    module.signals = module.signals || {}
```

Comment:
Also seems redundant. Would like to remove it and just secure controller.getSignal instead